packages/paste-website/src/components/icons-list/IconsList.tsx

```diff
@@ -0,0 +1,145 @@
+import {Heading} from '@twilio-paste/heading';
+import {IconTile} from './IconTile';
+import {IconObject, IconComponent, IconsListProps, GroupedList} from './types';
+import {IconCard} from './IconCard';
+import {getStickyColumnComputedOffset} from '../../utils/stickyUtils';
+import {SiteBodyContext} from '../site-wrapper/SiteBody';
+
+const iconsJson = require('@twilio-paste/icons/json/icons.json');
+
+const IconComponents: IconComponent = iconsJson.reduce((icons: IconComponent, {name}: IconObject) => {
+  return {
+    ...icons,
+    // eslint-disable-next-line import/no-dynamic-require,global-require
+    [`${name}`]: require(`@twilio-paste/icons/esm/${name}`)[name],
+  };
+}, {});
+
+const getGroupedList = (icons: IconsListProps['icons']): GroupedList =>
+  icons.reduce(
+    (prev: GroupedList, current): GroupedList => {
+      const Icon = {...current, component: IconComponents[current.name]};
+      if (current.name.startsWith('Product')) {
+        return {
+          products: [...prev.products, Icon],
+          ui: [...prev.ui],
+        };
+      }
+      return {
+        products: [...prev.products],
+        ui: [...prev.ui, Icon],
+      };
+    },
+    {products: [], ui: []}
+  );
+
+const getFirstIcon = (iconsList: GroupedList): IconObject | null => {
+  if (iconsList.ui.length !== 0) {
+    return iconsList.ui[0];
+  }
+  if (iconsList.products.length !== 0) {
+    return iconsList.products[0];
+  }
+  return null;
+};
+
+const IconTileWrapper = styled(Box)({
+  width: '33.33%',
+});
+
+const IconColumn = styled(Box)({
+  width: '41.66666666666667%',
+});
+
+const IconsList: React.FC<IconsListProps> = () => {
+  const {isPasteTheme} = React.useContext(SiteBodyContext);
+  const filterID = useUID();
+  const iconKeySeed = useUIDSeed();
+  const groupedIconList = getGroupedList(iconsJson);
+  const [filterString, setFilterString] = React.useState('');
+  const [iconsList, setIconsList] = React.useState(groupedIconList);
+  const [selectedIcon, setSelectedIcon] = React.useState(getFirstIcon(iconsList));
+  return (
+    <Box maxWidth="size120">
+      <Grid gutter="space60">
+        <Column span={7}>
+          <Box marginBottom="space70">
+            <FormLabel htmlFor={filterID}>Filter Icons</FormLabel>
+            <FormInput
+              id={filterID}
+              type="text"
+              value={filterString}
+              placeholder="Filter by name..."
+              onChange={e => {
+                const filteredList = iconsJson.filter(({name}: IconObject) =>
+                  name.toLowerCase().includes(e.currentTarget.value.toLowerCase())
+                );
+                const filteredGroupedList = getGroupedList(filteredList);
+                setIconsList(filteredGroupedList);
+                setSelectedIcon(getFirstIcon(filteredGroupedList));
+                setFilterString(e.currentTarget.value);
+              }}
+            />
+          </Box>
+        </Column>
+        <Column span={7}>
```

Comment:
7 + 7 = 14, which is bigger than 12.
Typo?

Reply:
Intentional for this current layout. I want the filter input container to be the same width as the column of icons below it, so it doesn't stretch 100%.